src/main/python/mainapp.py

```diff
@@ -1,22 +1,33 @@
 from fbs_runtime.application_context.PyQt5 import ApplicationContext
+
+from PyQt5.QtCore import QObject, pyqtSignal
+
 from app import App
+from api import Api
 from login import Login
-from PyQt5.QtCore import QObject, pyqtSignal
 
 
 class MainApp(QObject):
     login_signal = pyqtSignal()
     logout_signal = pyqtSignal()
 
-    def __init__(self, cxt: ApplicationContext, dont_boot_for_test=True, *args, **kwargs):
+    def __init__(
+        self, cxt: ApplicationContext, dont_boot_for_test=True, *args, **kwargs
+    ):
         super(QObject, self).__init__(*args, **kwargs)
         self.dont_boot_for_test = dont_boot_for_test
         self.login = None
         self.app = None
 
         self.cxt = cxt
 
-        self.show_login()
+        with Api("/auth/refresh") as api:
```

Comment:
is this the main change here?  I'll add the comments to the api repo for this change I guess...

Reply:
Yep, this is the main change. (I apologize, I didn't realize this branch had previous commits on it : /)